userstats.php

```diff
@@ -0,0 +1,67 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Prints a particular instance of moodleoverflow
+ *
+ * You can have a rather longer description of the file as well,
+ * if you like, and it can span multiple lines.
+ *
+ * @package   mod_moodleoverflow
+ * @copyright 2023 Tamaro Walter
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+// Include config and locallib.
+require_once(__DIR__.'/../../config.php');
+global $CFG, $PAGE, $DB, $OUTPUT, $SESSION;
+require_once($CFG->dirroot.'/mod/moodleoverflow/locallib.php');
+
+use mod_moodleoverflow\tables\userstats_table;
+// Declare optional parameters.
+$cmid = required_param('id', PARAM_INT);             // Course Module ID.
+$courseid = required_param('courseid', PARAM_INT);   // Course ID.
+$mid = required_param('mid', PARAM_INT);             // Moodleoveflow ID, Moodleoverflow that started the statistics.
+
```

Comment:
the check for viewanyrating is already down in the locallib.php:
https://github.com/learnweb/moodle-mod_moodleoverflow/blob/feature/userstats/locallib.php#L137
and only if canseeuserstats is true, the teacher can access the userstats

Reply:
done*

Reply:
You still need to repeat the check, as a user might "randomly" insert the url without having the capability!

Reply:
but doesnt the require.php do a moodle internal check? That was an issue that we looked up last week.
Or can I do a moodle internal in the userstats.php? It would make sense, since the userstats.php site is only available through the button

Reply:
ähh the require_once()...